models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -299,8 +299,8 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
             // sorry for the goto/label - but in this case it makes life really easier and the code more readable
             prepareTransformations:
 
-            foreach ($transformations as $transformation) {
-                if (!empty($transformation)) {
+            foreach ($transformations as &$transformation) {
+                if (!empty($transformation) && !$transformation['isApplied'] ?? true) {
```

Comment:
```suggestion
                if (!empty($transformation) && !$transformation['isApplied'] ?? false) {
```
I guess that's the issue